### PR TITLE
EmpeddedMongoDB plugin

### PR DESCRIPTION
**Pull request by rfelcman**
Main reason for this extension is that current plugin `com.github.joelittlejohn.embedmongo:embedmongo-maven-plugin` used in the EclipseLink project doesn't seem to be active now and doesn't support latest MongoDB version which should leads into problems to use and verify `org.mongodb:mongodb-driver-sync` instead of non-maintained `org.mongodb:mongo-java-driver`.
See EclipseLink PR [MongoDB Java driver upgrade](https://github.com/eclipse-ee4j/eclipselink/pull/1989).
There are integration tests and GitHub action added.

### CODE REVIEW

File: el-embedmongo-plugin/src/main/java/org/eclipse/persistence/embedmongo/StartMojo.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+import static org.apache.commons.lang3.StringUtils.equalsIgnoreCase;
+import static org.codehaus.plexus.util.StringUtils.contains;
+
+/**
+ * When invoked, this goal starts an instance of mongo. The required binaries
+ * are downloaded if no mongo release is found in <code>~/.embedmongo</code>.
+ *
+ */
+@Mojo(name="start", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST)
```
Comment (lukasj) [inline el-embedmongo-plugin/src/main/java/org/eclipse/persistence/embedmongo/StartMojo.java]: use `start-mongo` or sth similar

File: el-maven-plugin/src/main/java/org/eclipse/persistence/build/embedmongo/log/Loggers.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                .build();
+    }
+
+    public static ProcessOutput console() {
+        return ProcessOutput.namedConsole("");
+    }
+
+    public static ProcessOutput none() {
+        NoopStreamProcessor noop = new NoopStreamProcessor();
+        return ProcessOutput.silent();
```
Comment (lukasj) [inline el-maven-plugin/src/main/java/org/eclipse/persistence/build/embedmongo/log/Loggers.java:40]: this looks strange - `noop` is unused

File: el-maven-plugin/src/main/java/org/eclipse/persistence/build/embedmongo/StartMojo.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+import static org.apache.commons.lang3.StringUtils.equalsIgnoreCase;
+import static org.codehaus.plexus.util.StringUtils.contains;
+
+/**
+ * When invoked, this goal starts an instance of mongo. The required binaries
+ * are downloaded if no mongo release is found in <code>~/.embedmongo</code>.
+ *
+ */
+@Mojo(name="start-mongo", defaultPhase = LifecyclePhase.PRE_INTEGRATION_TEST)
```
Comment (lukasj) [inline el-maven-plugin/src/main/java/org/eclipse/persistence/build/embedmongo/StartMojo.java]: is this threadsafe? take a look at parallel builds

Comment (rfelcman) [reply el-maven-plugin/src/main/java/org/eclipse/persistence/build/embedmongo/StartMojo.java]: It's not threadsafe. Added explicit `, threadSafe = false` . Maybe not needed as `org.apache.maven.plugins.annotations.Mojo` has `boolean threadSafe() default false;`

File: el-maven-plugin/src/test/it/allPropertiesTest/pom.xml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    <groupId>org.eclipse.persistence</groupId>
+    <artifactId>eclipselink-testbuild-plugin-embedmongo-all-properties-test</artifactId>
+    <version>1.0-SNAPSHOT</version>
+
+    <name>basic-test</name>
+    <url>https://github.com/eclipse-ee4j/eclipselink-build-support</url>
+
+    <properties>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
```
Comment (lukasj) [inline el-maven-plugin/src/test/it/allPropertiesTest/pom.xml]: usage of `maven.compiler.release`/`maven.compiler.testRelease` is preferred

**Review by lukasj — APPROVED** [review]

**Pull request merged**